packages/rest/src/spec-enhancers/consolidate.spec-enhancer.ts

```diff
@@ -0,0 +1,182 @@
+// Copyright IBM Corp. 2020. All Rights Reserved.
+// Node module: @loopback/rest
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {
+  ApplicationConfig,
+  bind,
+  BindingScope,
+  CoreBindings,
+  inject,
+} from '@loopback/core';
+import {
+  asSpecEnhancer,
+  ISpecificationExtension,
+  isSchemaObject,
+  OASEnhancer,
+  OpenApiSpec,
+  ReferenceObject,
+  SchemaObject,
+} from '@loopback/openapi-v3';
+import debugFactory from 'debug';
+import compare from 'json-schema-compare';
+import _ from 'lodash';
+
+const debug = debugFactory('loopback:openapi:spec-enhancer:consolidate');
+
+/**
+ * This enhancer consolidates schemas into `/components/schemas` and replaces
+ * instances of said schema with a $ref pointer.
+ *
+ * Please note that the title property must be set on a schema in order to be
+ * considered for consolidation.
+ *
+ * For example, with the following schema instance:
+ *
+ * ```json
+ * schema: {
+ *   title: 'loopback.example',
+ *   properties: {
+ *     test: {
+ *       type: 'string',
+ *     },
+ *   },
+ * }
+ * ```
+ *
+ * The consolidator will copy the schema body to
+ * `/components/schemas/loopback.example` and replace any instance of the schema
+ * with a reference to the component schema as follows:
+ *
+ * ```json
+ * schema: {
+ *   $ref: '#/components/schemas/loopback.example',
+ * }
+ * ```
+ *
+ * When comparing schemas to avoid naming collisions, the description field
+ * is ignored.
+ */
+@bind(asSpecEnhancer, {scope: BindingScope.SINGLETON})
+export class ConsolidationEnhancer implements OASEnhancer {
+  name = 'consolidate';
+  disabled: boolean;
+
+  constructor(
+    @inject(CoreBindings.APPLICATION_CONFIG, {optional: true})
+    readonly config?: ApplicationConfig,
+  ) {
+    this.disabled = !(this.config?.rest?.openApiSpec?.consolidate || true);
```

Comment:
I think this won't work as expected. When `consolidate` is set to `false, the condition `consolidate || true` will be resolved as `true`. I am surprised this was not caught by the test suite 🤔 

Did you perhaps mean this?

```suggestion
    this.disabled = !(this.config?.rest?.openApiSpec?.consolidate ?? true);
```

Reply:
Hey @bajtos, that was a magnificent catch!  I did not create adequate tests.  I think I need a bit of help with them.  See [last commit](https://github.com/strongloop/loopback-next/pull/4365/commits/3fe2df25b8b900a6aa7c68dae8bf8b749cc2d677#diff-ed9321f709f68e00fe664ccf139a3ca6R422).

Reply:
Uh oh, I am afraid I don't have bandwidth to look into this in detail. Perhaps one of the OpenAPI owners can help?

Ping @jannyHou (primary owner) and @deepakrkris @emonddr (standby owners).

Reply:
@dougal83 the nullish coalescing looks correct to me. I see you also added two tests for the flag in https://github.com/strongloop/loopback-next/pull/4365#discussion_r410273651, better to have a 3rd check for the default value `true`.

Reply:
I couldn't figure out why the path is returned as an empty object. Please see error : https://travis-ci.com/github/strongloop/loopback-next/jobs/320173171#L468